analyzers/src/SonarAnalyzer.Common/Helpers/KnownReference.Predicates.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers;
+
+public sealed partial class KnownReference
+{
+    internal static Func<AssemblyIdentity, bool> NameIs(string name) =>
+        x => x.Name.Equals(name);
+
+    internal static Func<AssemblyIdentity, bool> StartsWith(string name) =>
+        x => x.Name.StartsWith(name);
+
+    internal static Func<AssemblyIdentity, bool> EndsWith(string name) =>
+        x => x.Name.EndsWith(name);
```

Comment:
Do you have a clear example in mind? Experience with KnownType shows that we typically exactly know what we're looking for.
```suggestion
```

Reply:
For StartsWith I was thinking of something like [Azure.Storage](https://www.nuget.org/packages?q=Azure.Storage&packagetype=&frameworks=&tfms=&prerel=True&sortby=relevance), [Azure.ResourceManager](https://www.nuget.org/packages?q=Azure.ResourceManager&packagetype=&frameworks=&tfms=&prerel=True&sortby=relevance), [AWSSDK](https://www.nuget.org/packages?q=AWSSDK+&packagetype=&frameworks=&tfms=&prerel=True&sortby=relevance) or [Microsoft.Extensions](https://www.nuget.org/packages?q=Microsoft.Extensions&packagetype=&frameworks=&tfms=&prerel=True&sortby=relevance) and for Contains something like [IdentityModel.Tokens](https://www.nuget.org/packages?q=IdentityModel.Tokens) or [SqlClient](https://www.nuget.org/packages?q=SqlClient&packagetype=&frameworks=&tfms=&prerel=True&sortby=relevance) or [Logger](https://www.nuget.org/packages?q=Logger&packagetype=&frameworks=&tfms=&prerel=True&sortby=relevance)

Reply:
There are also checks like "IsTestProject" or "Is this targeting .Net Core" or "Is this a WebProject" or "Is this a cloud app". These shortcuts can come in handy for those.